lib/OpenQA/Worker/Job.pm

```diff
@@ -385,10 +386,14 @@ sub _stop_step_3_announce ($self, $reason, $callback) {
 }
 
 sub _redact_file ($file_path, $file_name) {
-    eval { redact_settings_in_file($file_path) if $file_name eq 'vars.json' };
-    my $error = $@;
-    log_warning "Skipping upload of $file_name because an error occurred when stripping credentials: $error" if $error;
-    return !$error;
+    try {
+        redact_settings_in_file($file_path) if $file_name eq 'vars.json';
+        return 1;
+    }
+    catch ($e) {
+        log_warning "Skipping upload of $file_name because an error occurred when stripping credentials: $e";
+        return 0;
+    }
```

Comment:
```suggestion
    try { return redact_settings_in_file($file_path) if $file_name eq 'vars.json' }
    catch ($e) {
        log_warning "Skipping upload of $file_name because an error occurred when stripping credentials: $e";
        return 0;
    }
```